[docs]: Improve Installation page under Getting Started (#18037)
expo/expo: docs/pages/get-started/installation.md

import { Terminal } from '~/ui/components/Snippet';

> To develop applications with Expo, you need two tools. A command-line application called [Expo CLI](#1-expo-cli) to initialize and serve your project, and a mobile client app called [Expo Go](#2-expo-go-app-for-ios-and) to open the project on iOS and Android platforms. You can use any web browser to run the project on the web.

Reviewer: As we're no longer using `expo-cli` for initialization, I wonder if this should just say 'serve'?

Author: Great point. I forgot about the `create-expo-app` 😅 I have changed it.

> You don't need macOS to build an iOS app with Expo. You only need an iOS device to run the Expo Go app. Windows, Linux, and macOS are all supported for your development machine.

## 1. Expo CLI

> [Expo CLI](../workflow/expo-cli.md) is a command-line app that is the primary interface between a developer and Expo tools. It has a web-based Graphical User Interface (GUI) that pops up in the web browser when you start your project. You can use the GUI instead of the command line interface if you are not yet comfortable using a terminal or prefer GUIs. Both have similar capabilities.

Reviewer: > It has a web-based Graphical User Interface (GUI) that pops up in the web browser when you start your project etc. etc.

That is already disabled for new users and will be removed completely shortly, so I wouldn't mention that here anymore.

Author: Didn't know about it. Thanks for letting me know. I have removed it in the latest commit.

> [Expo CLI](../workflow/expo-cli.md) is a command-line app that is the primary interface between a developer and Expo tools. It has a web-based Graphical User Interface (GUI) that pops up in the web browser when you start your project. You can use the GUI instead of the command line interface if you are not yet comfortable using a terminal or prefer GUIs. Both have similar capabilities.

Reviewer 2: ```suggestion
[Expo CLI](/workflow/expo-cli) is a command-line app that is the primary interface between a developer and Expo tools. It has a web-based Graphical User Interface (GUI) that pops up in the web browser when you start your project. You can use the GUI instead of the command line interface if you are not yet comfortable using a terminal or prefer GUIs. Both have similar capabilities.
```

Just a suggestion - relative links tends to break more easily when moving content or pages, so I generally like to use the full path. Also you do not need to include `.md` in URL, this is a legacy requirement.

Author: Thanks for letting me know. I think the https://github.com/expo/expo/tree/main/docs#internal-linking is what confused me a little about the paths. I will see to it and send a PR to update that as well.

Reviewer 2: Oh, I didn't know we have doc on that. IIRC `yarn lint-links` check is manually disabled right now, but I don't remember what was the reason to skip this.

Author: Ah okay, please keep in loop about `yarn lint-links`. If that is something deprecated or removed, I will update the Readme file accordingly.

### Requirements

> With Expo CLI, you need to have the following tools installed on your developer machine:

Reviewer: 'With' as in 'Besides', or 'Together with in order to use'?

Author: I think "To use Expo CLI" as a phrase, sounds more accurate here. It sets a clear instruction that without having these other tools installed, Expo CLI will not work.

Author: I'll change it.

- [Node.js LTS release](https://nodejs.org/en/)
- [Git](https://git-scm.com)

> - [Watchman](https://facebook.github.io/watchman/docs/install#buildinstall) required only for macOS or Linux users

Reviewer: ```suggestion
- [Watchman](https://facebook.github.io/watchman/docs/install#buildinstall), required only for macOS or Linux users
```

> > Only Node.js LTS releases (even-numbered) are recommended. As Node.js [officially states](https://nodejs.org/en/about/releases/), "Production applications should only use Active LTS, or Maintenance LTS releases."

Reviewer: Don't add the comma, it's a quote, and their page doesn't have it.

Author: Okay, will revert this. Thanks!

### Recommended tools

- [VS Code Editor](https://code.visualstudio.com/download)
  - [VS Code Expo Extension](https://marketplace.visualstudio.com/items?itemName=byCedric.vscode-expo) for **app.json** debugging and autocomplete
- [Yarn](https://classic.yarnpkg.com/en/docs/install)
- Windows users: [PowerShell](https://docs.microsoft.com/en-us/powershell/scripting/install/installing-powershell-core-on-windows), Bash via WSL, or the VS Code terminal

### Installing Expo CLI

> Expo CLI is installed as a global npm package from a terminal app on your development machine. You will use it for different tasks in the development life-cycle of your project.

Reviewer: ```suggestion
Expo CLI is installed as a global npm package from a terminal app on your development machine. You will use it for different tasks in the development life cycle of your project.
```

<Terminal cmd={[
  '# Install the command line tools',
  '$ npm install --global expo-cli'
]} />

To verify the successful installation of CLI, run the following command:

<Terminal cmd={['$ expo whoami']} />

If the installation is successful, you will see a "Not logged in" message since you are not logged in to an Expo account yet. You do not need an account to start and can proceed further with your project. However, if you want to register a new expo account, run the command:

<Terminal cmd={['$ expo register']} />

> If you already have an Expo account, you can log into it by running the command:

Reviewer: ```suggestion
If you already have an Expo account, you can log in to it by running the command:
```

<Terminal cmd={['$ expo login']} />

> > **Need help?** Try searching the [forums](https://forums.expo.dev) &mdash; which is a great resource for troubleshooting.

Reviewer: Why 'is'?

Author: Forums is a single place/website. To be grammatical correct, it can either be `... are great resources for troubleshooting.` or `which is a great resource for troubleshooting.`. LMK if you think the former sounds more grammatically accurate. I will update this line.

Reviewer: Yeah, I'd prefer the first, 'forums' is plural, regardless of whether it indicates a single place (but I'd agree on the edge case aspect of it)

Author: Reverted in new commits. Kept the plural form.

## 2. Expo Go app for iOS and Android

The fastest way to get up and running is to use the Expo Go app on your iOS or Android device. Expo Go allows you to open up apps served through Expo CLI. It is available on both the iOS App Store and Android Play Store.

- [Android Play Store](https://play.google.com/store/apps/details?id=host.exp.exponent) - Android Lollipop (5) and greater.
- [iOS App Store](https://apps.apple.com/app/expo-go/id982107779) - iOS 11 and greater.

> Open the Expo Go app after it has finished installing. If you have created an account with `expo-cli`, you can sign in within the app from the "Profile" tab. Signing in will make it easier for you to open projects in the Expo Go app while developing them &mdash; they will appear automatically in the "Projects" tab of the app.

Reviewer: ```suggestion
Open the Expo Go app after it has finished installing. If you have created an account with `expo-cli`, you can sign in from the "Profile" tab. Signing in will make it easier for you to open projects in the Expo Go app while developing them &mdash; they will appear automatically in the "Projects" tab of the app.
```
Sentence starts with 'open the app', so the profile tab is already implied to be inside the app.
